File: hackerrank/array/easy/java_solutions/EqualizeTheArray.java
package hackerrank.array.easy.java_solutions;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class EqualizeTheArray {

    // https://www.hackerrank.com/challenges/equality-in-a-array/problem

    static class Solution {
        public static int equalizeArray(List<Integer> arr) {
            Map<Integer, Integer> m = new HashMap<>();

            int maxVal = 0, maxNum = 0;

            for (Integer v : arr) {
                int counter = m.getOrDefault(v, 0) + 1;
                m.put(v, counter);
                if (counter > maxVal) {
                    maxVal = counter;
                    maxNum = v;
                }
            }
            return arr.size() - m.get(maxNum);
        }
    }
}

File: hackerrank/array/easy/golang_solutions/equalize_the_array.go
package golang_solutions

// https://www.hackerrank.com/challenges/equality-in-a-array/problem

func EqualizeArray(arr []int32) int32 {
	// Write your code here
	m := make(map[int32]int)

	maxVal := 0
	maxNum := int32(0)

	for _, v := range arr {
		m[v]++
		if m[v] > maxVal {
			maxNum = v
			maxVal = m[v]
		}
	}
	return int32(len(arr) - m[maxNum])
}
